pkg/backends/calico/client.go

```diff
@@ -839,3 +856,39 @@ func (c *client) updateLogLevel() {
 		logutils.SetLevel("info")
 	}
 }
+
+var routeKeyPrefix = "/calico/staticroutes/"
+
+func (c *client) updateRoutes(cidrs []string) {
+
+	// Convert the slice of CIDRs to a map with corresponding
+	// cache keys.
+	routeMap := map[string]string{}
+	for _, cidr := range cidrs {
+		routeMap[routeKeyPrefix+strings.Replace(cidr, "/", "-", 1)] = cidr
+	}
+
+	// Lock the cache.
+	c.cacheLock.Lock()
+	defer c.cacheLock.Unlock()
+
+	// Reconcile the new route map against the cache.
+	for k, _ := range c.routeCache {
+		_, ok := routeMap[k]
+		if !ok {
+			// This cache entry should be deleted.
+			delete(c.routeCache, k)
```

Comment:
is it okay to delete an entry in a map while iterating over it?

might it be safer to reconcile the maps by generating a new one and then setting it to `c.routeCache`?

Reply:
Yes, it is safe; please look for "is removed during iteration" here: https://golang.org/ref/spec#RangeClause